### PR TITLE
WIP: Cherry-Pick changes from AOSP

### PR DESCRIPTION
**Pull request by LeFrosch**
This PR pulls all changes from the AOSP until [3e7c8ecfb0ccdc4fbe9e1bdfa862f2b267e87b0a](https://cs.android.com/android-studio/platform/tools/adt/idea/+/3e7c8ecfb0ccdc4fbe9e1bdfa862f2b267e87b0a). Also this PR starts to prepare the intellij fork for receiving further commits from the AOSP in future.

### CODE REVIEW

File: aspect/fast_build_info.bzl
Comment (mai93) [inline aspect/fast_build_info.bzl:1]: I know this is removed in AOSP but I think we still need it for Java development in intellij

File: third_party/bazel/src/main/protobuf/analysis_v2.proto
Comment (mai93) [inline third_party/bazel/src/main/protobuf/analysis_v2.proto:1]: For `third_part/bazel`, we should keep what we have as we are getting the same changes bazel team makes to these proto files while the code in AOSP has an older version of these files.

File: shared/java/com/google/idea/blaze/common/Label.java [hunk 1]
```diff
@@ -46,19 +48,20 @@ public static Label of(String label) {
     return interner.intern(new Label(label));
   }
 
-  public static Label fromPackageAndName(Path packagePath, Path name) {
-    return of(String.format("@@//%s:%s", packagePath, name));
+  public static Label fromWorkspacePackageAndName(String workspace, Path packagePath, Path name) {
+    return workspace.isEmpty() ? Label.of(String.format("//%s:%s", packagePath, name))
+        : Label.of(String.format("@@%s//%s:%s", workspace, packagePath, name));
```
Comment (LeFrosch) [inline shared/java/com/google/idea/blaze/common/Label.java]: @tpasternak Just rebased your most recent changes. Is this correct?

**Review by ujohnny — APPROVED** [review]
let's merge it as rebase

**Comment by tpasternak** [comment]
btw you can disable AS checks in the same PR

**Pull request merged**